### ISSUE TITLE
Corda Docker Implementation Timeout Issue

### ISSUE THREAD

**Issue by shashankkshukla** [corda/corda#5072]
While trying to create Corda docker instance faced connection timeout issue. Please share your views as to what is going wrong.

> **Step 1 - Notary Config File**
Placed Notary_node.conf file on CordaDockerTest directory 

```
devMode=true
myLegalName="O=Notary,L=Washington,C=US"
notary {
    validating=false
}
p2pAddress="localhost:10200"
rpcSettings {
    address="localhost:10201"
    adminAddress="localhost:10202"
}
rpcUsers=[
    {
        password=test
        permissions=["ALL"]-v /home/shashank/Handcraft/CordaDockerTest/Notary/cordapps:/opt/corda/cordapps \
	-v /home/shashank/Handcraft/sharednetwork/additional-node-infos:/opt/corda/additional-node-infos \
	-v /home/shashank/Handcraft/sharednetwork/network-parameters:/opt/corda/network-parameters \
        -p 10001:10200 \
        -p 10002:10201 \
        -p 10003:10202 \
        corda/corda-zulu-5.0-snapshot:latest
        user=user1
    }
]

```
> **Generating Notary Node**
Using this command all required corda node files got generated successfully.

`java -jar ../corda-tools-network-bootstrapper-4.0.jar --dir CordaDockerTest`

> **Docker Command**

```
docker run -ti \
	-v /home/shashank/Handcraft/CordaDockerTest/Notary/config:/etc/corda \
	-v /home/shashank/Handcraft/CordaDockerTest/Notary/certificates:/opt/corda/certificates \
	-v /home/shashank/Handcraft/CordaDockerTest/Notary/persistence:/opt/corda/persistence \
	-v /home/shashank/Handcraft/CordaDockerTest/Notary/logs:/opt/corda/logs \
	-v /home/shashank/Handcraft/CordaDockerTest/Notary/cordapps:/opt/corda/cordapps \
	-v /home/shashank/Handcraft/sharednetwork/additional-node-infos:/opt/corda/additional-node-infos \
	-v /home/shashank/Handcraft/sharednetwork/network-parameters:/opt/corda/network-parameters \
        -p 10001:10200 \
        -p 10002:10201 \
        -p 10003:10202 \
        corda/corda-zulu-5.0-snapshot:latest
```

> **Downloaded Client Test Code**
`https://github.com/corda/corda-tut1-solution-java`

> **Modified the Configuration in build.gradle**
```
task runTemplateClient(type: JavaExec) {
    classpath = sourceSets.main.runtimeClasspath
    main = 'com.template.Client'
    args 'localhost:10002', 'user1', 'test'
}

```
> **Executed Command**
`./gradlew runTemplateClient`
> **Error Faced**
> Task :clients:runTemplateClient
I 16:40:11 1 RPCClient.logElapsedTime - Failed Startup took 31589 msec
Exception in thread "main" ActiveMQConnectionTimedOutException[errorType=CONNECTION_TIMEDOUT message=AMQ119013: Timed out waiting to receive cluster topology. Group:null]
        at org.apache.activemq.artemis.core.client.impl.ServerLocatorImpl.createSessionFactory(ServerLocatorImpl.java:814)
        at net.corda.client.rpc.internal.RPCClientProxyHandler.start(RPCClientProxyHandler.kt:216)
        at net.corda.client.rpc.internal.RPCClient$start$1.invoke(RPCClient.kt:93)
        at net.corda.client.rpc.internal.RPCClient$start$1.invoke(RPCClient.kt:32)
        at net.corda.core.internal.InternalUtils.logElapsedTime(InternalUtils.kt:193)
        at net.corda.core.internal.InternalUtils.logElapsedTime(InternalUtils.kt:183)
        at net.corda.client.rpc.internal.RPCClient.start(RPCClient.kt:72)
        at net.corda.client.rpc.CordaRPCClient.start(CordaRPCClient.kt:392)
        at net.corda.client.rpc.CordaRPCClient.start(CordaRPCClient.kt:375)
        at net.corda.client.rpc.CordaRPCClient.start(CordaRPCClient.kt:344)
        at com.template.Client.main(Client.java:33)
> **Note**

> When the same Notary node is started directly i.e. without docker container the client code listed above executes successfully.

**Comment by shashankkshukla** [comment]
This issue is resolved after adding --network=host --hostname=127.0.0.1 in docker run command.

**Comment by shashankkshukla** [comment]
This issue is resolved after adding --network=host --hostname=127.0.0.1 in docker run command.
